### ISSUE TITLE
Paused VMI shouldn't be marked as ready

### ISSUE THREAD

**Issue by zcahana** [kubevirt/kubevirt#5920]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

A VMI with no readiness probe has its `Ready: True` condition set when it's paused.

**What you expected to happen**:

A paused VMI shouldn't be marked as ready, regardless of the readiness probe it defines.

**How to reproduce it (as minimally and precisely as possible)**:

```sh
# Create a VM with no readiness probe.
$ kubectl apply -f cirros.yaml

# Wait for the VMI to be running
$ ...

# Pause the VMI
$ virtctl pause vmi cirros

# List VMI and see its Ready condition
$ kubectl get vmi cirros -o jsonpath="{.status.conditions[?(@.type=='Ready')].status}

**Comment by phoracek** [comment]
In general this sounds right, but there may be some consequences to this change we don't see now. Marking as accepted, but would probably merit a design document/considerations - to prove it does not break any existing functionality.

**Comment by phoracek** [comment]
@kwiesmueller @rthallisey

**Comment by zcahana** [comment]
@phoracek can you please elaborate what such consequences do you have in mind?
I reckon that a proof for not breaking existing functionality should arise when the fix (which I think should be relatively straightforward) passes unit and functional tests.

**Comment by phoracek** [comment]
Some concerns about whether this is a bug, and whether change in the behavior would affect the rest of KubeVirt, were raised during the community meeting. I can't speak for test coverage. I would leave it to @kwiesmueller who is far more familiar with this part of codebase than I am.

**Comment by rmohr** [comment]
I think `virt-controller` could make use of https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate and set the pod to not-ready while the VM is paused, if no readiness probes are set.

**Comment by zcahana** [comment]
This is being addressed in #6533.

/remove-lifecycle stale
/assign
